nexus/src/authz/update.polar

```diff
@@ -0,0 +1,31 @@
+# Base update case: any version trivially updates to itself.
+update(_component, from, from, []);
+
+# Unit tests: update anything trivially, but not arbitrarily.
+?= update("foo", "foo", "foo", []);
+?= not update("foo", "foo", "bar", []);
+
```

Comment:
It does seem like this would be a really nice format to ensure compatibility between version boundaries - especially if we have the language for:

- Describing general compatibility windows (from version XXX, you can update to anything in the range of YYY-ZZZ).
- Describing "gated" changes (you must update to version XXX before you can update to version YYY)
- Describing "point-of-no-return" changes (after you update to version XXX, you can no longer downgrade to version YYY)

This also seems like a really nice format for ensuring some degree of liveness, for example:

- Only update components which have redudancy elsewhere (e.g., a single scrimlet at a time, and only do so when the other one is online)
- Prioritize components with smaller workloads